examples/evm_bots.py

```diff
@@ -402,33 +207,37 @@ def set_up_experiment(
     }
 
     # dynamically load devnet addresses from address file
-    address_file = None
-    if args.devnet:
-        addresses, address_file = get_devnet_addresses(experiment_config, addresses)
-    return pricing_model, crash_file, network_choice, provider_settings, addresses, address_file
+    if args["devnet"]:
+        addresses= get_devnet_addresses(experiment_config, args, addresses)
+    return pricing_model, crash_file, network_choice, provider_settings, addresses
 
-
-def get_devnet_addresses(experiment_config: Config, addresses: dict[str, str]) -> tuple[dict[str, str], str]:
+def get_devnet_addresses(experiment_config: Config, args: dict, addresses: dict[str, str]) -> tuple[dict[str, str], str]:
     """Get devnet addresses from address file."""
-    address_file = experiment_config.scratch["project_dir"] / "artifacts" / "addresses.json"
-    # make parent folder if it doesn't exist
-    os.makedirs(os.path.dirname(address_file), exist_ok=True)
-    address_file_data = {}
-    if Path.exists(address_file):
-        with open(address_file, "r", encoding="utf-8") as file:
-            address_file_data = json.load(file)
-    if "baseToken" in address_file_data:
-        addresses["baseToken"] = address_file_data["baseToken"]
+    deployed_addresses = {}
+    # get deployed addresses from local file, if it exists
+    address_file_path = experiment_config.scratch["project_dir"] / "hyperdrive_solidity/artifacts/addresses.json"
+    if os.path.exists(address_file_path):
+        with open(address_file_path, "r", encoding="utf-8") as file:
+            deployed_addresses = json.load(file)
+    else:  # otherwise get deployed addresses from artifacts server
```

Comment:
we need to be aware that the existence of the file overrides use of the artifacts server. not sure this is the best pattern going forward, but it works for now.